routers/api/v1/repo/issue.go

```diff
@@ -834,6 +835,70 @@ func EditIssue(ctx *context.APIContext) {
 	ctx.JSON(http.StatusCreated, convert.ToAPIIssue(issue))
 }
 
+func DeleteIssue(ctx *context.APIContext) {
+	// swagger:operation DELETE /repos/{owner}/{repo}/issues/{index} issue issueDelete
+	// ---
+	// summary: Delete an issue
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: index
+	//   in: path
+	//   description: index of issue to delete
+	//   type: integer
+	//   format: int64
+	//   required: true
+	// responses:
+	//   "204":
+	//     "$ref": "#/responses/empty"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+	//   "404":
+	//     "$ref": "#/responses/notFound"
+
+	deleteIssue(ctx)
+}
+
+func deleteIssue(ctx *context.APIContext) {
+	if !ctx.IsSigned || !ctx.Repo.IsAdmin() {
```

Comment:
IIRC `reqToken` will error/fail when user isn't logged in, so no need to have the `IsSigned` check. As for the `ctx.Repo.IsAdmin` check. I think it's better to use the `reqAdmin` function within the routes to do this check.

Reply:
As someone not very familiar with the Gitea codebase, I'm not sure what you are talking about here. The code is as-is in similar functions. So you propose to change this line to
```suggestion
	if !ctx.reqToken() || !ctx.reqAdmin() {
```
or rather to

```suggestion
ctx.reqAdmin()
```
without the if statement?

Reply:
See other comment.